### PR TITLE
Add tests for `svm/transaction_processor.rs`

### PR DESCRIPTION
**Pull request by LucasSte**
#### Problem

Many functions in `svm/transaction_processor.rs` were missing tests. 

#### Summary of Changes

I created the pertinent tests. More will come in other PRs.

### CODE REVIEW

File: svm/src/transaction_processor.rs [hunk 1]
```diff
@@ -819,10 +820,6 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
             Some(account) => account,
         };
 
-        debug_assert!(solana_bpf_loader_program::check_loader_id(
-            program_account.owner()
-        ));
-
```
Comment (dmakarov) [inline svm/src/transaction_processor.rs]: what was the point of moving this assert down?

File: svm/src/transaction_processor.rs [hunk 2]
```diff
@@ -861,6 +862,7 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
                 );
             }
         }
+
```
Comment (dmakarov) [inline svm/src/transaction_processor.rs]: I'd prefer not adding empty lines, but it's ok to leave it in.

**Review by dmakarov — DISMISSED** [review]
LGTM

File: svm/src/transaction_processor.rs (hunk 1 above)
Comment (LucasSte) [reply svm/src/transaction_processor.rs]: If this assert is here, `if loader_v4::check_id(program_account.owner())` (see the line below) will never be true.

Comment (dmakarov) [reply svm/src/transaction_processor.rs]: Maybe a fix should be to add `loader_v4::check_id()` to `solana_bpf_loader_program::check_loader_id` instead?

Comment (LucasSte) [reply svm/src/transaction_processor.rs]: Added

File: svm/src/transaction_processor.rs (hunk 2 above)
Comment (LucasSte) [reply svm/src/transaction_processor.rs]: Removed

**Review by dmakarov — APPROVED** [review]

**Pull request merged**